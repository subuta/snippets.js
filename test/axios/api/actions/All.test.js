/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import IndexAction from 'lib/axios/api/actions/All'

import {
  Routes as RoutesConfig,
  Models as ModelsConfig
} from 'test/fixtures/config'

describe('axios/api/actions/All', () => {
  it('should create axios Api index action with channel', () => {
    const channelRoute = RoutesConfig['channel']
    const channelModel = ModelsConfig['channel']

    expect(format(IndexAction(channelRoute, channelModel))).toMatchSnapshot()
  })

  it('should create axios Api index action with comment', () => {
    const channelRoute = RoutesConfig['comment']
    const channelModel = ModelsConfig['comment']

    expect(format(IndexAction(channelRoute, channelModel))).toMatchSnapshot()
  })
})
