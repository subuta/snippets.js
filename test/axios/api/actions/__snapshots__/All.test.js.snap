// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`axios/api/actions/All should create axios Api index action with channel 1`] = `
"export const index = () => {
  return request.get(\`/channels\`)
}
"
`;

exports[`axios/api/actions/All should create axios Api index action with comment 1`] = `
"export const index = (params) => {
  const {channelId} = params
  return request.get(\`/channels/\${channelId}/comments\`)
}
"
`;
